main.go

```diff
@@ -551,115 +579,177 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+				e.parseAndNewMetric(ch, e.lruCrawlerSleep, prometheus.GaugeValue, settings, "lru_crawler_sleep"),
+				e.parseAndNewMetric(ch, e.lruCrawlerMaxItems, prometheus.GaugeValue, settings, "lru_crawler_tocrawl"),
+				e.parseBoolAndNewMetric(ch, e.lruMaintainerThread, prometheus.GaugeValue, settings, "lru_maintainer_thread"),
+				e.parseAndNewMetric(ch, e.lruHotPercent, prometheus.GaugeValue, settings, "hot_lru_pct"),
+				e.parseAndNewMetric(ch, e.lruWarmPercent, prometheus.GaugeValue, settings, "warm_lru_pct"),
+				e.parseAndNewMetric(ch, e.lruHotMaxAgeFactor, prometheus.GaugeValue, settings, "hot_max_factor"),
+				e.parseAndNewMetric(ch, e.lruWarmMaxAgeFactor, prometheus.GaugeValue, settings, "warm_max_factor"),
+			)
+			if err != nil {
+				parseError = err
+			}
+		}
+	}
+	return parseError
+}
+
+func (e *Exporter) parseAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parse, stats, key, labelValues...)
+}
+
+func (e *Exporter) parseBoolAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parseBool, stats, key, labelValues...)
+}
+
+func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, f func(map[string]string, string) (float64, error), stats map[string]string, key string, labelValues ...string) error {
+	v, err := f(stats, key)
+	if err != nil {
+		return err
 	}
+
+	ch <- prometheus.MustNewConstMetric(desc, valueType, v, labelValues...)
+	return nil
 }
 
-func parse(stats map[string]string, key string) float64 {
-	v, err := strconv.ParseFloat(stats[key], 64)
+func parse(stats map[string]string, key string) (float64, error) {
+	value, ok := stats[key]
+	if !ok {
+		log.Errorf("Key not found: %s", key)
+		return math.NaN(), errors.New("key not found")
+	}
+
+	v, err := strconv.ParseFloat(value, 64)
 	if err != nil {
-		log.Errorf("Failed to parse %s %q: %s", key, stats[key], err)
-		v = math.NaN()
+		log.Errorf("Failed to parse %s %q: %s", key, value, err)
+		return math.NaN(), err
 	}
-	return v
+	return v, nil
 }
 
-func parseBool(stats map[string]string, key string) float64 {
-	switch stats[key] {
+func parseBool(stats map[string]string, key string) (float64, error) {
+	value, ok := stats[key]
+	if !ok {
+		log.Errorf("Key not found: %s", key)
+		return math.NaN(), errors.New("key not found")
+	}
+
+	switch value {
 	case "yes":
-		return 1
+		return 1, nil
 	case "no":
-		return 0
+		return 0, nil
 	default:
-		log.Errorf("Failed parse %s %q", key, stats[key])
-		return math.NaN()
+		log.Errorf("Failed parse %s %q", key, value)
+		return math.NaN(), errors.New("failed parse a bool value")
```

Comment:
It'd be more idiomatic to return 0 for the float64, as that's the default value, instead of returning NaN.